includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -1565,6 +1565,22 @@ private function process_link_element( DOMElement $element ) {
 	 * @return string|WP_Error Stylesheet string on success, or WP_Error on failure.
 	 */
 	private function get_stylesheet_from_url( $stylesheet_url ) {
+		// For absolute paths, provide the origin (host and port).
+		if ( '/' === substr( $stylesheet_url, 0, 1 ) && '//' !== substr( $stylesheet_url, 0, 2 ) ) {
```

Comment:
I just tried disabling this condition and then attempted to run the test to see if the test was asserting the expected case, but the test still passed when this code was disabled. 😕 

Looking further, I'm not sure this logic is needed given this logic in `get_validated_url_file_path`:

https://github.com/ampproject/amp-wp/blob/85a264447b36fa68793dcf55a5602fe3e0639174/includes/sanitizers/class-amp-style-sanitizer.php#L1292-L1299

Reply:
Hmm. Additionally, I can attest that `get_validated_url_file_path` precisely covers file path validation. So long as there are no use cases that it cannot handle, we can rely on it and avoid adding additional logic for the same task.